sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -661,7 +666,9 @@ class Analyzer(
           val evaluatedOrderings = resolvedAliasedOrdering.zip(sortOrder).map {
             case (evaluated, order) =>
               val index = originalAggExprs.indexWhere {
-                case Alias(child, _) => child semanticEquals evaluated.child
+                case a@Alias(child, _) =>
+                  (child semanticEquals evaluated.child) ||
+                    a.exprId == evaluated.child.asInstanceOf[AttributeReference].exprId
                 case other => other semanticEquals evaluated.child
```

Comment:
I think our optimizer is smart enough to remove unnecessary `Project`, so we don't need to worry about it here :)
you can change your test case if the plan doesn't match.